src/bundle.c

```diff
@@ -879,6 +879,57 @@ static gboolean decrypt_bundle_payload(RaucBundle *bundle, GError **error)
 	return res;
 }
 
+static gboolean check_workdir_content(const gchar *path, GError **error)
+{
+	GError *ierror = NULL;
+	g_autofree gchar *name = g_path_get_basename(path);
+
+	/* It's not clear what realative symlinks in the bundle input should mean,
```

Comment:
```suggestion
	/* It's not clear what absolute symlinks (or relative symlinks that contain slashes) in the bundle input should mean,
```

Reply:
And we also need to actually copy the symlinks...

Reply:
It turns out that symlinks can be hard-linked just fine, so there is no need to special case that. I've mentioned this in the commit message.